src/container.jl

```diff
@@ -62,13 +56,8 @@ function forkr(trace::Trace)
     newf = reset_model(trace.f)
     Random123.set_counter!(trace.rng, 1)
 
-    ctask = let f = trace.ctask.task.code
-        Libtask.CTask() do
-            res = f()(trace.rng)
-            Libtask.produce(nothing)
-            return res
-        end
-    end
+    func = Libtask.func(trace.ctask)
+    ctask = Libtask.CTask(func, trace.rng)
```

Comment:
Maybe create a convenient constructor `Libtask.CTask(t::Ctask, args...)` so we don't need to handle `func` explicitly.

Reply:
Done.